test/Table_Tests/src/Common_Table_Operations/Derived_Columns_Spec.enso

```diff
@@ -197,3 +197,10 @@ spec setup =
             t.set (Column_Operation.Add (Column_Ref.Name "X") (Column_Ref.Name "Z")) . should_fail_with No_Such_Column
             t.set (Column_Operation.Not (Column_Ref.Name "zzz")) . should_fail_with No_Such_Column
             t.set (Column_Operation.Not (Column_Ref.Index 42)) . should_fail_with Index_Out_Of_Bounds
+
+    Test.group "Unique derived column names" <|
+        Test.specify "Should disambiguate two derived columns that would otherwise have had the same name" <|
+            t = table_builder [["X", [1, 2, 3]]]
+            column_op = Column_Operation.Power 2 (Column_Ref.Name "X")
+            t2 = t.set column_op . set column_op
+            t2.column_names . should_equal ["X", "[Constant_2] ^ [X]", "[Constant_2] ^ [X] 1"]
```

Comment:
Please also add a test where 2 constants are used _in a single expression_. That is the case I was worried about that could not work in SQL under some circumstances.

I think it will work, but let's have a test covering it.

e.g.
```
table.set "2 + (2 * 2) + (2 * [X])"
```
I'd like to verify what is the generated name, I guess something like `([2] + ([2] * [2])) + ([2] * [X])` but that is not all.

I'd like to test that `. to_vector` returns correct values in this case.
Just checking the column names will not run the actual SQL query - and we need to make sure that the generated query is without defects (that could _potentially_ come from clashing sub-expression names) and runs fine.

Reply:
Done